test/rubygems/test_gem_bundler_version_finder.rb

```diff
@@ -80,6 +80,7 @@ def test_bundler_version_with_reason
 
   def test_deleted_directory
     skip "Cannot perform this test on windows" if win_platform?
+    skip "Cannot perform this test on Solaris" if /solaris/ =~ RUBY_PLATFORM
```

Comment:
Can we make a `solaris_platform?` helper method here?

Reply:
Currently, this is only one case where the Solaris check is needed.  IMO, it would be good to refactor it out after the same check is required in multiple places.  But I'll update my PR if it is really needed.

Reply:
I'd like a helper method here too, but since this commit [has already landed ruby's master](https://github.com/ruby/ruby/commit/1b3339528c0804aa0e673bd3f15be8b087d17bd8) I don't mind merging this to get in sync first, and then refactoring separately.

Reply:
No further feedback, so let's do this now and extract the helper method later.